### PR TITLE
ILM: Get policy support wildcard name

### PR DESCRIPTION
**Pull request by weizijun**
ILM Get policy support wildcard name is a useful feature. 
We can use `Get  /_ilm/policy/policy*` to get the wildcard result.

### CODE REVIEW

**Comment by weizijun** [comment]
@dakrone can you help to reivew this PR?

**Comment by dakrone** [comment]
@weizijun supporting both `,` and wildcards leads to a lot of very difficult edge cases around response codes. I think for now it would be better to stick with *either* comma-separated specific values *or* a wildcard. (So either `GET /_ilm/policy/mypolicy1,mypolicy2` *OR* `GET /_ilm/policy/mypolicy*`.) Would you be able to make that change to this PR?

**Comment by weizijun** [comment]
> @weizijun supporting both `,` and wildcards leads to a lot of very difficult edge cases around response codes. I think for now it would be better to stick with _either_ comma-separated specific values _or_ a wildcard. (So either `GET /_ilm/policy/mypolicy1,mypolicy2` _OR_ `GET /_ilm/policy/mypolicy*`.) Would you be able to make that change to this PR?

Okay, I will change as you said. @dakrone ,do you mean that `GET /_ilm/policy/mypolicy1*,mypolicy2*` is failed? Or don't use wildcard to handle the two values?

I see that the old template action, `TransportGetIndexTemplatesAction` support multi names, and each name support wildcard. The new template action supports only one name. Which one is reasonable?

**Comment by dakrone** [comment]
> do you mean that `GET /_ilm/policy/mypolicy1*,mypolicy2*` is failed? Or don't use wildcard to handle the two values?

It should throw a helpful exception that says to use *either* comma-separated values *or* wildcards.

> I see that the old template action, `TransportGetIndexTemplatesAction` support multi names, and each name support wildcard. The new template action supports only one name. Which one is reasonable?

The new one is more reasonable. The old one supports it just because we cannot make the change without breaking backwards compatibility.

**Comment by weizijun** [comment]
hi, @dakrone , I have finished the change, can you help to review it?

**Comment by dakrone** [comment]
@elasticmachine ok to test

File: x-pack/plugin/ilm/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/ilm/10_basic.yml [hunk 1]
```diff
@@ -58,6 +58,96 @@ setup:
...
+
+  - do:
+      ilm.get_lifecycle:
+        policy: "my_timeseries*"
+  - match: { my_timeseries_lifecycle_1.version: 1 }
+  - match: { my_timeseries_lifecycle_2.version: 1 }
+
+  - do:
+      ilm.get_lifecycle:
+        policy: "my_timeseries_lifecycle_1, my_timeseries_lifecycle_2"
```
Comment (dakrone) [inline x-pack/plugin/ilm/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/ilm/10_basic.yml:132]: I believe this extra space is causing test failures:
```suggestion
        policy: "my_timeseries_lifecycle_1,my_timeseries_lifecycle_2"
```

**Review by dakrone — COMMENTED** [review]
@weizijun could you add a changelog file to this PR? I left a couple of other comments also

**Comment by dakrone** [comment]
@elasticmachine update branch

**Comment by dakrone** [comment]
The CI failures for this are transient and due to the recent 8.5.0 version bump, so I expect them to go away as those get sorted out.

**Comment by weizijun** [comment]
@elasticmachine update branch

**Review by dakrone — APPROVED** [review]
LGTM, thanks @weizijun!

**Comment by weizijun** [comment]
> LGTM, thanks @weizijun!

Thanks @dakrone !

**Pull request merged**